File: CodeChef/CHEFSTON.cpp
#include<bits/stdc++.h>
using namespace std;
#define max(a,b) (a>b)?a:b
int main(){
	int t,n,k;
	scanf("%d",&t);
	while(t--){
		int ans=0;
		scanf("%d %d",&n,&k);
		vector<int> a(n);
		vector<int> b(n);
		for(int i=0;i<n;i++)scanf("%d",&a[i]);
		for(int i=0;i<n;i++)scanf("%d",&b[i]);
		for(int i=0;i<n;i++)ans=max(ans,k/a[i]*b[i]);
		printf("%d\n",ans);
	}
	return 0;
}

File: CodeChef/CHEFSTON.py
T=int(input())
for t in range(T):
    n,k=map(int,input().split(" "))
    A=list(map(int,input().split(" ")))
    B=list(map(int,input().split(" ")))
    ans=0
    for i in range(n):
        ans=max(ans,(k//A[i] * B[i]))
    print(ans)
